catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+
+import ddf.catalog.Constants;
+import ddf.catalog.transform.InputTransformer;
+
+/**
+ * Provides information on available transformers
+ */
+@Service
+@Command(scope = CatalogCommands.NAMESPACE, name = "transformers", description = "Provides information on available transformers.")
+public class TransformersCommand extends CatalogCommands {
+
+    private static final String MIME_TYPE = "mime-type";
+
+    private static final String SCHEMA = "schema";
+
+    private static final String NOT_AVAILABLE = "N/A";
+
+    private static final String LINE = "------------------------";
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        List<TransformerProperties> transformersProperties = new ArrayList<>();
+
+        // ServiceReferences are needed to get property information from InputTransformers
+        List<ServiceReference<InputTransformer>> serviceReferences =
+                (ArrayList<ServiceReference<InputTransformer>>) bundleContext.getServiceReferences(
+                        InputTransformer.class,
+                        null);
+
+        for (ServiceReference ref : serviceReferences) {
+            String id = getTransformerProperty(ref, Constants.SERVICE_ID);
+
+            if (!id.equals(NOT_AVAILABLE)) {
+                String schema = getTransformerProperty(ref, SCHEMA);
+                List<String> mimeType = getTransformerMimeProperty(ref);
+
+                transformersProperties.add(new TransformerProperties(id, schema, mimeType));
```

Comment:
✏️ You can make the ctor for `TransformerProperties` smarter, passing in the `ServiceReference` and letting it extract the fields it needs. Then you can replace this loop with something simpler like...
```java
List<TransformerProperties> transformersProperties = serviceReferences.stream()
    .map(TransformerProperties::new)
    .collect(Collectors.toList());
```